lib/FusionInventory/Agent/Tools.pm

```diff
@@ -121,36 +121,45 @@ sub getCanonicalManufacturer {
         HygonGenuine => 'Hygon'
     );
 
+    my %regexp = (
+        "Apple"           => qr/^APPLE/i,
+        "Elpida"          => qr/^EB[EJ]/,
+        "Hewlett-Packard" => qr/^(hp|HP|(?i)hewlett[ -]packard)/,
+        "Hitachi"         => qr/^(HD|IC|HU|HGST)/,
+        "Hynix"           => qr/^(?:HMT|HMA|00AD0)/,
+        "Micron"          => qr/^002C0/,
+        "Samsung"         => qr/^00CE0/,
+        "Seagate"         => qr/^(ST|(?i)seagate)/,
+        "Sony"            => qr/^OPTIARC/i,
+        "Toshiba"         => qr/^THNSF/,
+        "Western Digital" => qr/^(WDC|(?i)western)/,
+    );
+
     if (exists $manufacturers{$manufacturer}) {
         $manufacturer = $manufacturers{$manufacturer};
     } elsif ($manufacturer =~ /(
-        maxtor     |
-        sony       |
+        \blg\b     |
         compaq     |
-        ibm        |
-        toshiba    |
         fujitsu    |
-        \blg\b     |
-        samsung    |
-        nec        |
-        transcend  |
-        matshita   |
         hitachi    |
-        pioneer
+        ibm        |
+        intel      |
+        matshita   |
+        maxtor     |
+        nec        |
+        pioneer    |
+        samsung    |
+        sony       |
+        toshiba    |
+        transcend
     )/xi) {
         $manufacturer = ucfirst(lc($1));
-    } elsif ($manufacturer =~ /^(hp|HP|(?i)hewlett[ -]packard)/) {
-        $manufacturer = "Hewlett-Packard";
-    } elsif ($manufacturer =~ /^(WDC|(?i)western)/) {
-        $manufacturer = "Western Digital";
-    } elsif ($manufacturer =~ /^(ST|(?i)seagate)/) {
-        $manufacturer = "Seagate";
-    } elsif ($manufacturer =~ /^(HD|IC|HU|HGST)/) {
-        $manufacturer = "Hitachi";
-    } elsif ($manufacturer =~ /^APPLE/i) {
-        $manufacturer = "Apple";
-    } elsif ($manufacturer =~ /^OPTIARC/i) {
-        $manufacturer = "Sony";
+    } else {
+        for (keys %regexp) {
+            $manufacturer =~ $regexp{$_} || next;
+            $manufacturer = $_;
+            last;
+        }
```

Comment:
Can you here try the following ?
```suggestion
        my $match = first { $manufacturer =~ $regexp{$_} } keys(%regexp);
        $manufacturer = $match if $match;
```

Reply:
I believe `first` function is from `List::Util`

Reply:
Indeed I also imported it in Tools.pm (this file) as a convenient tool a while ago.